diff --git a/go1_gym/envs/base/legged_robot.py b/go1_gym/envs/base/legged_robot.py
index 03bc6d9..8f2a087 100755
--- a/go1_gym/envs/base/legged_robot.py
+++ b/go1_gym/envs/base/legged_robot.py
@@ -1133,6 +1133,11 @@ class LeggedRobot(BaseTask):
         """
         # get gym GPU state tensors
         actor_root_state = self.gym.acquire_actor_root_state_tensor(self.sim)
+        actor_indices = list(range(actor_root_state.shape[0]))
+        num_actors = self.gym.get_sim_actor_count(self.sim)
+        actual_actor_indices = [i for i in actor_indices if i % num_actors == 0]
+        actor_root_state = actor_root_state
+        
         dof_state_tensor = self.gym.acquire_dof_state_tensor(self.sim)
         net_contact_forces = self.gym.acquire_net_contact_force_tensor(self.sim)
         rigid_body_state = self.gym.acquire_rigid_body_state_tensor(self.sim)
@@ -1143,7 +1148,8 @@ class LeggedRobot(BaseTask):
         self.gym.render_all_camera_sensors(self.sim)
 
         # create some wrapper tensors for different slices
-        self.root_states = gymtorch.wrap_tensor(actor_root_state)
+        breakpoint()
+        self.root_states = gymtorch.wrap_tensor(actor_root_state)[actual_actor_indices, :]
         self.dof_state = gymtorch.wrap_tensor(dof_state_tensor)
         self.net_contact_forces = gymtorch.wrap_tensor(net_contact_forces)[:self.num_envs * self.num_bodies, :]
         self.dof_pos = self.dof_state.view(self.num_envs, self.num_dof, 2)[..., 0]
